ch4 - QUICK PR - randomization rules (#320)
worldbank/d4di: chapters/sampling-randomization-power.tex

Basically, it has a really long ordered list of numbers with the property that
knowing the previous one gives you precisely zero information about the next one.
Stata uses one of these numbers every time it has a task that is non-deterministic.

> In ordinary use, it will cycle through these numbers starting from a new point

Reviewer: ```suggestion
In ordinary use, it will cycle through these numbers starting from a fixed point
```

Weirdly, this is correct as is. https://www.stata.com/manuals/rsetseed.pdf

> The sequences the random-number functions produce are determined by the seed, which is just a number and which is set to 123456789 every time Stata is launched. This means that runiform() produces the same sequence each time you start Stata.

Author: Ok, interesting. I did not know that and that seems like a very bad idea. #stata I will make sure it gets reverted and then I will close the comment.

  \url{https://dimewiki.worldbank.org/wiki/Master_Do-files}}
However, note that testing your do-files without running them
via the master do-file may produce different results,

> since Stata's \texttt{version} expires after each time you run your do-files.

Reviewer: ```suggestion
since Stata's \texttt{version} setting expires after each time you run your do-files.
```

\textbf{Sorting} means that the actual data that the random process is run on is fixed.

> Because numbers are assigned to each observation in row-by-row starting from 

Reviewer: ```suggestion
Because numbers are assigned to each observation row-by-row starting from 
```

In Stata, \texttt{set seed [seed]} will set the generator to that state.
You should use exactly one seed per randomization process.
The most important thing is that each of these seeds is truly random,
so do not use shortcuts such as the current date or a seed you have used before.
You will see in the code below that we include the source and timestamp for verification.
Any process that includes a random component is a random process,
including sampling, randomization, power calculation, and algorithms like bootstrapping.
Other commands may induce randomness in the data or alter the seed without you realizing it,
so carefully confirm exactly how your code runs before finalizing it.\sidenote{
  \url{https://dimewiki.worldbank.org/wiki/Randomization_in_Stata}}

To confirm that a randomization has worked well before finalizing its results,
save the outputs of the process in a temporary location,
re-run the code, and use \texttt{cf} or \texttt{datasignature} to ensure 

> nothing has changed. It is also advisable to let someone else re-produce your 

Reviewer: ```suggestion
nothing has changed. It is also advisable to let someone else reproduce your 
```

randomization results on their machine to remove any doubt that your results 

> are reproducable.

Reviewer: ```suggestion
are reproducible.
```
